drivers/counter/counter_ll_stm32_rtc.c

```diff
@@ -186,6 +191,170 @@ static inline void ll_func_disable_alarm(RTC_TypeDef *rtc)
 
 static void rtc_stm32_irq_config(const struct device *dev);
 
+/* When no error occurs, this function disables the RTC write protection and should be balanced
+ * with a call to rtc_stm32_exit_init_mode (which enables RTC write protection).
+ * In case of error, the write protection is enabled when leaving this function, so nothing more
+ * needs to be made.
+ */
+static int rtc_stm32_enter_init_mode(void)
+{
+#if defined(CONFIG_SOC_SERIES_STM32F1X)
+	/* Wait for RTC to be ready */
+	if (!WAIT_FOR(LL_RTC_IsActiveFlag_RTOF(RTC), RTC_TIMEOUT, NULL)) {
+		return -ETIMEDOUT;
+	}
+
+	LL_RTC_DisableWriteProtection(RTC);
+#else
+	LL_RTC_DisableWriteProtection(RTC);
+
+	/* Check if the Initialization mode is set */
+	if (LL_RTC_IsActiveFlag_INIT(RTC) == 0U) {
+		/* Set the Initialization mode */
+		LL_RTC_EnableInitMode(RTC);
+		if (!WAIT_FOR(LL_RTC_IsActiveFlag_INIT(RTC), RTC_TIMEOUT, NULL)) {
+			LL_RTC_EnableWriteProtection(RTC);
```

Comment:
`LL_RTC_DisableInitMode(RTC)` seems missing here. Sorry I missed that in my previous review.